src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1186,6 +1188,21 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
             }
         }
 
+        private bool CanApplyParseOptionChange(ParseOptions oldOptions, ParseOptions newOptions, Project project)
```

Comment:
@jasonmalinowski Latest commit takes a stab at this, but I wasn't sure the best way to split this method across layers. Thanks